org.eclipse.scanning.sequencer/src/org/eclipse/scanning/sequencer/nexus/SolsticeScanMonitor.java

```diff
@@ -118,6 +118,13 @@ public void setNexusObjectProviders(Map<ScanRole, List<NexusObjectProvider<?>>>
 			.collect(Collectors.toList());  // collect in a list
 
 		this.nexusObjectProviders = nexusObjectProviderList;
+		
+		final List<NexusObjectProvider<?>> detectors = nexusObjectProviderMap.get(ScanRole.DETECTOR);
+		// we can write the unique key for each position on move if all detectors write their own unique key
+		// TODO what about monitors?
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)